src/NumericalAlgorithms/RootFinding/NewtonRaphson.hpp

```diff
@@ -89,4 +90,58 @@ DataVector newton_raphson(const Function& f, const DataVector& initial_guess,
   return result_vector;
 }
 
+/*!
```

Comment:
Let's hold off on adding something with this functionality until we have a more concrete example of where it will be used. (Cornell group code review)